### PR TITLE
[G&M] iModelId, Mapping and Group names in log messages

### PR DESCRIPTION
**Pull request by smmr-dn**
Removed iModelId, and replaced MappingId and GroupId with names in log messages.
mapping-names

### CODE REVIEW

File: packages/itwin/grouping-mapping-widget/src/widget/components/ExtractionMessageModal.tsx [hunk 1]
```diff
@@ -21,6 +24,42 @@ export interface ExtractionMessageModalProps {
...
   const [formattedTimestamp, setFormattedTimestamp] = useState<string>("");
+  const groupingMappingApiConfig = useGroupingMappingApiConfig();
+  const mappingClient = useMappingClient();
+  const [mappingId, setMappingId] = useState<string>("");
+  const [groupId, setGroupId]  = useState<string>("");
+  const { mappings } = useMappingsOperations({...groupingMappingApiConfig, mappingClient});
+  useEffect(() => {
+    extractionMessageData.forEach((extractionMessage) => {
+      if(extractionMessage.message.includes("iModel")){
+        extractionMessage.message = extractionMessage.message.replace(/iModel [\w-]+/, "iModel");
```
Comment (Guillar1) [inline packages/itwin/grouping-mapping-widget/src/widget/components/ExtractionMessageModal.tsx:35]: It's not a good idea usually to directly modify objects like this. You will have to recreate the object and set it again.

File: packages/itwin/grouping-mapping-widget/src/widget/components/ExtractionMessageModal.tsx [hunk 2]
```diff
@@ -21,6 +24,42 @@ export interface ExtractionMessageModalProps {
...
+  }, [mappingId, groupId, groupingMappingApiConfig, mappings, mappingClient]);
+  useEffect(() => {
+    extractionMessageData.forEach(async (extractionMessage) => {
+      if(extractionMessage.message.includes("MappingId:")){
+        const splittedMessage = extractionMessage.message.split(" ");
+        setMappingId(splittedMessage[splittedMessage.indexOf("MappingId:") + 1]);
+        setGroupId(splittedMessage[splittedMessage.indexOf("GroupingId:") + 1]);
+        await setMappingandGroupNames(extractionMessage);
+      }
+    });
```
Comment (Guillar1) [inline packages/itwin/grouping-mapping-widget/src/widget/components/ExtractionMessageModal.tsx:61]: You can probably wrap this in a `Promise.all`

File: packages/itwin/grouping-mapping-widget/src/widget/components/ExtractionMessageModal.tsx [hunk 3]
```diff
@@ -21,6 +24,42 @@ export interface ExtractionMessageModalProps {
...
+  const [mappingId, setMappingId] = useState<string>("");
+  const [groupId, setGroupId]  = useState<string>("");
+  const { mappings } = useMappingsOperations({...groupingMappingApiConfig, mappingClient});
+  useEffect(() => {
+    extractionMessageData.forEach((extractionMessage) => {
+      if(extractionMessage.message.includes("iModel")){
+        extractionMessage.message = extractionMessage.message.replace(/iModel [\w-]+/, "iModel");
+      }
+    });
+  }, [extractionMessageData]);
```
Comment (Guillar1) [inline packages/itwin/grouping-mapping-widget/src/widget/components/ExtractionMessageModal.tsx:38]: I feel like this useEffect can be combined with the other one.

File: packages/itwin/grouping-mapping-widget/src/widget/components/ExtractionMessageModal.tsx [hunk 4]
```diff
@@ -21,6 +25,51 @@ export interface ExtractionMessageModalProps {
...
+      const groups = await mappingClient.getGroups(
+        accessToken,
+        groupingMappingApiConfig.iModelId,
+        mappingId
+      );
+      groupsCache.set(mappingId, groups);
+    }
+    return groupsCache.get(mappingId)?.find((group) => {return group.id === groupId;})?.groupName ?? "";
+  }, [groupingMappingApiConfig, mappingClient]);
+  useEffect(() => {
```
[the remark this answers is not in the document]
Comment (smmr-dn) [reply packages/itwin/grouping-mapping-widget/src/widget/components/ExtractionMessageModal.tsx]: I just added those spaces.

File: packages/itwin/grouping-mapping-widget/src/components/Mappings/Extraction/ExtractionMessageModal.tsx [hunk 5]
```diff
@@ -20,7 +24,55 @@ export interface ExtractionMessageModalProps {
...
+      const extractionMessageDataPromises = extractionMessageData.map(async (extractionMessage) => {
+        let replacedMessage = extractionMessage.message;
+        if(extractionMessage.message.includes("iModel")){
+          replacedMessage = replacedMessage.replace(/iModel [\w-]+/, "iModel");
+        }
+        if(replacedMessage.includes("MappingId:")){
+          const splittedMessage = replacedMessage.split(" ");
+          const mappingId = splittedMessage[splittedMessage.indexOf("MappingId:") + 1];
+          const groupId = splittedMessage[splittedMessage.indexOf("GroupId:") + 1];
+          const mappingName = await getMappingName(mappingId);
```
Comment (Guillar1) [inline packages/itwin/grouping-mapping-widget/src/components/Mappings/Extraction/ExtractionMessageModal.tsx:62]: `getMappingNmae` doesn't need to be awaited.

Comment (smmr-dn) [reply packages/itwin/grouping-mapping-widget/src/components/Mappings/Extraction/ExtractionMessageModal.tsx]: I have had it removed.

**Review by Guillar1 — APPROVED** [review]

**Pull request merged**